### PR TITLE
{front-door} Bump up SDK for API 20201101 version

### PR DESCRIPTION
**Pull request by disreedh**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
front door

**Comment by kairu-ms** [comment]
@disreedh Please fix CI issues first. Thanks.

**Comment by disreedh** [comment]
@kairu-ms  Thank you for helping out with the issue.

File: src/front-door/azext_front_door/_params.py [hunk 1]
```diff
@@ -164,6 +164,8 @@ def load_arguments(self, _):
         c.argument('redirect_url', help='URL used for redirect rule action.')
         c.argument('custom_block_response_status_code', help='HTTP status to return for blocked requests.')
         c.argument('custom_block_response_body', help='Body to return for blocked requests.')
+        c.argument('request_body_check', arg_type=get_enum_type(PolicyRequestBodyCheck), help='Disabled or Enabled status. Default value is Disabled')
```
Comment (kairu-ms) [inline src/front-door/azext_front_door/_params.py:167]: PolicyRequestBodyCheck only have "Enabled" and "Disabled" values. Please use get_three_state_flag 

``` python
arg_type=get_three_state_flag(positive_label='Enabled', negative_label='Disabled', return_label=True)
```

File: src/front-door/azext_front_door/custom.py [hunk 2]
```diff
@@ -646,19 +646,22 @@ def update_fd_routing_rule(parent, instance, item_name, frontend_endpoints=None,
...
         tags=tags,
+        sku=Sku(name=sku if sku is not None else SkuName.classic_azure_front_door),
         policy_settings=PolicySettings(
             enabled_state='Enabled' if not disabled else 'Disabled',
             mode=mode,
             redirect_url=redirect_url,
             custom_block_response_status_code=custom_block_response_status_code,
-            custom_block_response_body=custom_block_response_body
+            custom_block_response_body=custom_block_response_body,
+            request_body_check=request_body_check if request_body_check is not None else PolicyRequestBodyCheck.disabled,
```
Comment (kairu-ms) [inline src/front-door/azext_front_door/custom.py:664]: get_three_state_flag will return disabled if use don't provide that parameter. You can see code for detail

File: src/front-door/azext_front_door/custom.py [hunk 3]
```diff
@@ -668,16 +671,21 @@ def create_waf_policy(cmd, resource_group_name, policy_name,
...
+    with UpdateContext(instance.sku) as c:
+        c.update_param('name', sku, SkuName.classic_azure_front_door)
+
     with UpdateContext(instance.policy_settings) as c:
         c.update_param('enabled_state', 'Enabled' if not disabled else 'Disabled', 'Disabled')
         c.update_param('mode', mode, False)
         c.update_param('redirect_url', redirect_url, None)
         c.update_param('custom_block_response_status_code', custom_block_response_status_code, None)
         c.update_param('custom_block_response_body', custom_block_response_body, None)
+        c.update_param('request_body_check', request_body_check, 'Disabled')
```
Comment (kairu-ms) [inline src/front-door/azext_front_door/custom.py:688]: If user already create a instance which request_body_check is enabled. And then user update that instance without request_body_check parameter. Should we set it to Disabled?

Comment (kairu-ms) [reply src/front-door/azext_front_door/custom.py:688]: same question for "enabled_state" and "mode" property

Comment (kairu-ms) [reply src/front-door/azext_front_door/custom.py:688]: I think we should keep the origin value, instead of Disable.

**Review by kairu-ms — CHANGES_REQUESTED** [review]

File: src/front-door/azext_front_door/_params.py (hunk 1 above)
Comment (disreedh) [reply src/front-door/azext_front_door/_params.py]: Replaced with get_three_state_flag

File: src/front-door/azext_front_door/custom.py (hunk 2 above)
Comment (disreedh) [reply src/front-door/azext_front_door/custom.py]: Sure, simplified it

File: src/front-door/azext_front_door/custom.py (hunk 3 above)
Comment (disreedh) [reply src/front-door/azext_front_door/custom.py]: I have changed request_body_check, mode, enabled_state and sku to keep origin value during update

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**